### ISSUE TITLE
Add setting to not automatically expand images, gists, etc.

### ISSUE THREAD

**Issue by lucaswerkmeister** [gitterHQ/gitter#714]
Some people dislike Gitter’s feature of automatically inlining and expanding gists, videos, and other things. It would be nice to have an option to *not* automatically expand these.

- @sadmac7000: https://gitter.im/ceylon/user?at=553dd491bbc455106370c1a7
- @jvasileff: https://gitter.im/ceylon/user?at=554b7c2fa5eca5f3705b7391
- @sadmac7000: https://gitter.im/ceylon/user?at=554fa224629c6099462539b0
- @quintesse: https://gitter.im/ceylon/user?at=55508082cb59bb0b439660d1
- @lucaswerkmeister: https://gitter.im/ceylon/user?at=5564b60a0d9fc84f46ba3f8d
- @sadmac7000: https://gitter.im/ceylon/user?at=556c178b9f3a9b7b060a155e
- @sadmac7000: https://gitter.im/ceylon/user?at=556e3e70777c17d06a13aa9f
- @tombentley: https://gitter.im/ceylon/user?at=557199ac05c872ce6ac807cf

**Comment by tombentley** [comment]
I find the full-sized nature of the inlining particularly annoying. I tend to spend a lot of time scrolling back in history and a few full sized videos, gists and images mean I even more time on the mouse wheel.

**Comment by dasher** [comment]
It should be a setting with a sensible default - and maybe it would be good to allow people to interact with the unexpanded item to see it in detail.
It's a nice feature when you actually wanna see it - but otherwise it's a bit of a pain.

**Comment by lucaswerkmeister** [comment]
@dasher there’s already a button to toggle between collapsed and expanded form. I’d expect that to still be present and functional. It would just start in “collapsed” state.

**Comment by quintesse** [comment]
Btw, I do actually like that things like images and videos show up in-line, I'd just expect them to be much smaller, they can expand or open in a new tab when clicked upon.
Links to gists I'd collapse by default indeed.

**Comment by gdejohn** [comment]
Yeah, I like having the ability to inline these kinds of things, but I'd really like the option to have them collapsed by default.

**Comment by MadLittleMods** [comment]
Just to document these related issues:

 - [add support for a collapse all-function #668](https://github.com/gitterHQ/gitter/issues/668)
 - [have an option to collapse all gifs #535](https://github.com/gitterHQ/gitter/issues/535)
 - [Collapse code blocks #515](https://github.com/gitterHQ/gitter/issues/515)

**Comment by SaintPeter** [comment]
+1 on this.  As per #919, a bad Codepen can also "kill" a room.

**Comment by abhisekp** [comment]
:+1: 
- Customizable category based **permanent** collapse/expand toggle setting where categories include
 - Coding
 - Videos
 - Images
 - etc
- Temporary Individual media toggle feature

**Comment by LeonineKing1199** [comment]
:+1: 

This would be an amazing quality of life improvement. As of now, gitter is amazing but it kind of hurts the site when people who don't know better post giant gists or trolls want to spam distracting gifs.

**Comment by shmutalov** [comment]
Year 2017...

**Comment by kbrock** [comment]
While configuration may be nice, I only have 2 issues:

1. Images sometimes take up the whole screen (other times they are perfect)
2. Gists are expanded by default. (would prefer they not be expanded ever)

I'm sure other people also only have 2 main issues, but they are probably different from my own.
